yt/yt/server/query_tracker/query_tracker.cpp

```diff
@@ -318,14 +318,9 @@ class TQueryTracker
             leaseTransactionId,
             optionalRecord->State);
 
-        // If current query state is "pending", switch it to "running". Otherwise, keep the existing state of a query;
-        // in particular, it may be "failing" or "completing" if the previous incarnation succeeded in reaching pre-terminating state.
-        auto newState = optionalRecord->State == EQueryState::Pending ? EQueryState::Running : optionalRecord->State;
-
         auto rowBuffer = New<TRowBuffer>();
         TActiveQueryPartial newRecord{
             .Key = queryRecord.Key,
-            .State = newState,
```

Comment:
Why do we remove it? Shouldn't we keep the same state?

`.State = optionalRecord->State`

Reply:
Here, we only update the specified fields, so "State" will remain the same if it is not specified.